Sources/ForageSDK/Component/ForagePANTextField/MaskedUITextField.swift

```diff
@@ -0,0 +1,202 @@
+//
+//  MaskedUITextField.swift
+//  
+//
+//  Created by Danilo Joksimovic on 2023-08-14.
+//
+
+import UIKit
+import Foundation
+
+extension String {
+    subscript(safe index: Int) -> Character? {
+        return index < count && index >= 0 ? self[self.index(startIndex, offsetBy: index)] : nil
+    }
+}
+
+internal enum MaskPattern : String {
+    case unset = "###################"
+    case sixteenDigits = "#### #### #### ####"
+    case eighteenDigits = "###### #### ##### ## #"
+    case nineteenDigits = "###### #### #### ### ##"
+}
+
+internal class MaskedUITextField : UITextField, ObservableState {
+    // MARK: - Properties
+    private var wasBackspacePressed = false
+    
+    /// A delegate that informs the client about the state of the entered card number (validation, focus)
+    public weak var forageDelegate: ForageElementDelegate? {
+        didSet {
+            delegate = forageDelegate as? UITextFieldDelegate
+        }
+    }
+    
+    @IBInspectable private(set) public var isEmpty = true
+    // TODO: default to isValid = true on initialization
```

Comment:
https://linear.app/joinforage/issue/FX-449/ios-update-default-isvalid-of-pin-and-pan-elements-to-be-true

Reply:
Default this val to true :)